### ISSUE TITLE
swiftlint analyzer with grouping extension

### ISSUE THREAD

**Issue by xuaninbox** [realm/SwiftLint#2389]
It seems the new analyzer lint does not apply to the grouping extension, will this be supported in the future?

```swift
import A
import B   <- Warning: B is not used

class Test {
    // Use A
}

extension Test {
    // Use B
}
```

**Comment by jpsim** [comment]
Can you explain what you mean by grouping extension? Are you saying that the `unused_import` rule is producing false positives in some cases? If so, can you share a sample project that triggers the issue? Thanks!

**Comment by niilohlin** [comment]
I just tried

```Swift
import CoreData // A
import UIKit // B

class Person {
    var context: NSManagedObject? // Use A
}

extension Person {
    var view: UIView { // Use B
        return UIView()
    }
}
```

And did not get a false positive.
